src/components/DraggableList/types.tsx

```diff
@@ -0,0 +1,22 @@
+import type {DraggableChildrenFn} from 'react-beautiful-dnd';
+import type {RenderItemParams as OriginalRenderItemParams} from 'react-native-draggable-flatlist';
+
+type DefaultItemProps = {
+    id: string;
+};
+
+type DraggableListProps<T extends DefaultItemProps> = {
+    data: T[];
+    keyExtractor: (item: T, index: number) => string;
+    renderItem: (params: RenderItemParams<T>) => React.ReactNode;
+    onDragEnd?: (params: {data: T[]}) => void;
```

Comment:
Could we extract this type `{data: T[]}`?

Reply:
I've extracted it, but I'm not sure if this is what you meant.

https://github.com/Expensify/App/blob/c974a3768b1777877cba48e5e92ca52b70de715b/src/components/DraggableList/types.ts#L4-L18

Reply:
Yes, is exactly that! But could you rename `DataType` to `OnDragEndParams`? It will make more sense in this context.

Reply:
I'm not sure if the `OnDragEndParams` name would work here as I also use this type [here](https://github.com/Expensify/App/blob/c974a3768b1777877cba48e5e92ca52b70de715b/src/components/DraggableList/types.ts#L17).

Maybe something like `DraggableListData` would be better? Or maybe I should remove this `& DataType<T>;` and use `data: T[]` as before.

cc @blazejkustra

Reply:
Oh yes, I didn't see `& DataType<T>;`.. I agree with `DraggableListData`!